### ISSUE TITLE
spotlight tojson does not export target, 3js does not edit it

### ISSUE THREAD

**Issue by makc** [mrdoob/three.js#9508]
```json
{
            "uuid": "F7A1537F-0DA9-4FC2-9436-1BCC2A930EDA",
            "type": "SpotLight",
            "castShadow": true,
            "matrix": [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 215.52757263183594, -383.2663269042969, 587.3773193359375, 1],
            "color": 255,
            "intensity": 2,
            "distance": 1300,
            "angle": 0.3490658503988659,
            "decay": 1,
            "penumbra": 0
        }```

this is what I get for the spot light. importing it in 3js editor it looks at the origin.

**Comment by tentone** [comment]
I can solve this probably by storing the uuid of the target object in the JSON file and replacing that object after all objects are loaded), wich would take an adicional pass in the objectloader, still this would not take care of the cases when we have the pre-initialized object that is created here:

SpotLight.js
`this.target = new Object3D();`

So this can be kinda of tricky

But i have a diferent propose wich is to change from an object target to a Vector3 target (you would still be able to look at an object position object), this target should be world coordinates.

**Comment by makc** [comment]
docs say you should add target object to the scene, so it will likely be exported as well

**Comment by tentone** [comment]
The problem is that you dont actually are required to do that, Hmmm, what do you think about moving this to a Vector3? Just as a possibility

**Comment by makc** [comment]
If the target can be anywhere in scene graph, vector3 cannot replace that?

**Comment by tentone** [comment]
It can as it is rigth now the target.wordmatrix, to avoid changing a lot of stuff we could make target as an object that stores position and a tranf. matrix.

And after saying this i think that its just plain better to leave it as it is and just serialize the target position and  target uuid.

And on load if no object that matches the uuid is found we can set the pre-initialized target object position to the stored one, wich is actually a small change.

**Comment by mrdoob** [comment]
The reason is that, eventually, I want to get rid of `target`...

**Comment by tentone** [comment]
Spot lights could just use object rotation and if somebody wants the light to follow something can use the lookAt function easily

**Comment by makc** [comment]
> I want to get rid of `target`...

could you at least make the matrix right then?

**Comment by mrdoob** [comment]
Oh, the matrix is broken?

**Comment by makc** [comment]
In the way that it does not look at the target - yes

**Comment by WestLangley** [comment]
>could you at least make the matrix right then?

`spotlight.quaternion` and `spotlight.rotation` are not used. Hence the rotation component of the matrix is irrelevant.

Spotlights act like point lights, but with a restricted cone of illumination.

**Comment by makc** [comment]
ok but if ever you going to lose the target, the matrix will be relevant. so it seems you could start implementing it by making serialization work?

**Comment by makc** [comment]
in fact, I do not suggest this is the definite solution. but this stuff is obviously not working, and you better fix it, in any possible way. OR ELSE! or else it will continue to be broken :D

**Comment by WestLangley** [comment]
>Spot lights could just use object rotation and if somebody wants the light to follow something can use the lookAt function easily

The problem is, `lookAt()` does not support objects with rotated a rotated parent.

**Comment by tentone** [comment]
Hmm thats true, sorry my bad. (Should that be fixed?)

Anyway i think just trowing the target away and use the light rotation would be kinda of easy anyway since the shader takes a direction vector its fast to calculate one from euler rotation.

**Comment by mrdoob** [comment]
Yep, the reason we haven't done this yet is because `ShadowMap` currently relies on having a target.
